packages/flutter/lib/src/services/text_input.dart

```diff
@@ -1425,3 +1552,55 @@ class TextInput {
     );
   }
 }
+
+/// An interface for attaching and detaching [TextInputClient].
+///
+/// [TextInputSource] creates is responsible for creating [TextInputConnection]
+/// instances that are used to communicate with the currently attached
+/// [TextInputClient].
+///
+/// The default text input source, available via [TextInput.defaultSource],
+/// creates [TextInputConnection] instances that communicate with the platform
+/// text input plugin over the [SystemChannels.textInput] method channel. See
+/// [SystemChannels.textInput] for more details about the text input method
+/// channel messages.
```

Comment:
super nit: I feel we can combine this paragraph and the dedicated "Custom text input sources" section since they're both example usages?